src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java

```diff
@@ -40,6 +41,51 @@
         return query.getResultList();
     }
 
+    /**
+     * Gets from the data base all the comments, grouped by student, of one section
+     * @param sectionId
+     * @return a map with pairing a student with all his/her comments
+     */
+    @Override
+    public Map<Long, String> getAttendanceCommentsBySectionId(long sectionId) {
```

Comment:
Any reason you cannot use jpql in this method? It's good to default to jpql unless raw sql is needed for speed.

Btw this might help. Jpql has a `new` keyword which enables you to load data into objects which are not entities.

`select new ExampleAttendanceReportsClass(att.studentId, att.dateOfClass, att.notes) from Attendance att...`